topology/src/test/java/io/camunda/zeebe/topology/changes/TopologyChangeCoordinatorImplTest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.topology.changes;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.atomix.cluster.MemberId;
+import io.camunda.zeebe.scheduler.future.ActorFuture;
+import io.camunda.zeebe.scheduler.testing.TestActorFuture;
+import io.camunda.zeebe.scheduler.testing.TestConcurrencyControl;
+import io.camunda.zeebe.topology.ClusterTopologyAssert;
+import io.camunda.zeebe.topology.ClusterTopologyManager;
+import io.camunda.zeebe.topology.changes.TopologyChangeCoordinatorImpl.InvalidTopologyChangeException;
+import io.camunda.zeebe.topology.changes.TopologyChangeCoordinatorImpl.OperationNotAllowed;
+import io.camunda.zeebe.topology.state.ClusterTopology;
+import io.camunda.zeebe.topology.state.MemberState;
+import io.camunda.zeebe.topology.state.TopologyChangeOperation;
+import io.camunda.zeebe.topology.state.TopologyChangeOperation.PartitionChangeOperation.PartitionJoinOperation;
+import io.camunda.zeebe.topology.state.TopologyChangeOperation.PartitionChangeOperation.PartitionLeaveOperation;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.function.UnaryOperator;
+import org.junit.jupiter.api.Test;
+
+final class TopologyChangeCoordinatorImplTest {
+
+  final TestClusterTopologyManager clusterTopologyManager = new TestClusterTopologyManager();
+
+  @Test
+  void shouldFailOnInvalidChanges() {
```

Comment:
:wrench: Add one more test to validate that the _second_ operation can still cause a failure, i.e. two leave operations for the same member.